### PR TITLE
Remove `*_url_list` filters

### PR DESCRIPTION
**Pull request by swissspidy**
### Issue Number
Part of #164

### Description
Depends on #183.
Depends by #191.

With those new filters, there isn't really a need for these filters here. They're even more harmful as they could mess with pagination.

### Type of change
Please select the relevant options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (change which improves an existing feature. E.g., performance improvement, docs update, etc.)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Steps to test
N/A

### Acceptance criteria
- [x] My code follows WordPress coding standards.
- [x] I have performed a self-review of my own code.
- [x] If the changes are visual, I have cross browser / device tested.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added test instructions that prove my fix is effective or that my feature works.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,60 +60,43 @@ add_filter(
...
-			}
-		}
-		return $urls;
-	},
-	10,
-	2
+	'wp_sitemaps_posts_query_args',
+	function( $args ) {
+		$args['post__not_in'] = isset( $args['post__not_in'] ) ? $args['post__not_in'] : array();
+		$args['post__not_in'][] 42;
```
Comment (pfefferle) [inline README.md]: `=` missing

File: README.md [hunk 2]
```diff
@@ -60,60 +60,43 @@ add_filter(
...
-			}
-		}
-		return $urls;
-	},
-	10,
-	2
+	'wp_sitemaps_taxonomies_query_args',
+	function( $args ) {
+		$args['exclude'] = isset( $args['exclude'] ) ? $args['exclude'] : array();
+		$args['exclude'][] 7;
```
Comment (pfefferle) [inline README.md]: `=` missing

File: README.md [hunk 3]
```diff
@@ -60,60 +60,43 @@ add_filter(
...
-		$user_to_remove = array( 'loc' => get_author_posts_url( 1 ) );
-		$key = array_search( $user_to_remove, $urls, true );
-		if ( false !== $key ) {
-			array_splice( $urls, $key, 1 );
-		}
-		return $urls;
+	'wp_sitemaps_users_query_args',
+	function( $args ) {
+		$args['exclude'] = isset( $args['exclude'] ) ? $args['exclude'] : array();
+		$args['exclude'][] 1;
```
Comment (pfefferle) [inline README.md]: `=` missing

File: readme.txt [hunk 4]
```diff
@@ -76,60 +76,43 @@ add_filter(
...
-			}
-		}
-		return $urls;
-	},
-	10,
-	2
+	'wp_sitemaps_posts_query_args',
+	function( $args ) {
+		$args['post__not_in'] = isset( $args['post__not_in'] ) ? $args['post__not_in'] : array();
+		$args['post__not_in'][] 42;
```
Comment (pfefferle) [inline readme.txt]: `=` missing

File: readme.txt [hunk 5]
```diff
@@ -76,60 +76,43 @@ add_filter(
...
-			}
-		}
-		return $urls;
-	},
-	10,
-	2
+	'wp_sitemaps_taxonomies_query_args',
+	function( $args ) {
+		$args['exclude'] = isset( $args['exclude'] ) ? $args['exclude'] : array();
+		$args['exclude'][] 7;
```
Comment (pfefferle) [inline readme.txt]: `=` missing

File: readme.txt [hunk 6]
```diff
@@ -76,60 +76,43 @@ add_filter(
...
-		$user_to_remove = array( 'loc' => get_author_posts_url( 1 ) );
-		$key = array_search( $user_to_remove, $urls, true );
-		if ( false !== $key ) {
-			array_splice( $urls, $key, 1 );
-		}
-		return $urls;
+	'wp_sitemaps_users_query_args',
+	function( $args ) {
+		$args['exclude'] = isset( $args['exclude'] ) ? $args['exclude'] : array();
+		$args['exclude'][] 1;
```
Comment (pfefferle) [inline readme.txt]: `=` missing

File: README.md (hunk 1 above)
Comment (swissspidy) [reply README.md]: good catch, thanks!

**Comment by pfefferle** [comment]
@swissspidy what about the `pre_` filter? should we use this MR to document these too?

**Comment by swissspidy** [comment]
No strong opinion. I can just make a new one for that.

**Review by pfefferle — APPROVED** [review]

**Pull request merged**